app/uk/gov/hmrc/pensionschemereturn/transformations/nonsipp/MemberPaymentsTransformer.scala

```diff
@@ -131,7 +131,9 @@ class MemberPaymentsTransformer @Inject()(
           memberDetails = details,
           employerContributionsDetails = SectionDetails(
             made = out.employerContributionMade.boolean,
-            completed = out.memberDetails.forall(_.noOfContributions.nonEmpty)
+            completed =
+              if (!out.employerContributionMade.boolean) true
+              else !out.memberDetails.forall(_.noOfContributions.isEmpty)
```

Comment:
As a matter of interest why not `out.memberDetails.forall(_.noOfContributions.nonEmpty)` ?

Reply:
The reason the original code isn't working as intended is because when we try and POST a value of `Some(0)`, log out and log back in, then GET the submitted details, the value for `noOfContributions` has been discarded / treated as `None`. In this case,`.nonEmpty` will evaluate as false, and what this means is that, if at least 1 member has 0 contributions, then the Status will always evaluate as In Progress, any time the user logs back in.